base/server/cms/src/com/netscape/cms/servlet/tks/SecureChannelProtocol.java

```diff
@@ -2346,5 +2395,69 @@ public SymmetricKey generateAESSymKey(String selectedToken, int keySize) throws
 
         return symKey;
     }
+    private static byte[] debugAESKeyToBytes(CryptoToken token,SymmetricKey aesKey) {
+        KeyGenerator kg;
+        SymmetricKey sessionKey;
+        byte[] result = null;
+
+        if(token == null || aesKey == null) {
+            return result;
+        }
+
+        try {
+            kg = token.getKeyGenerator(KeyGenAlgorithm.AES);
+
+            SymmetricKey.Usage usages[] = new SymmetricKey.Usage[4];
+            usages[0] = SymmetricKey.Usage.WRAP;
+            usages[1] = SymmetricKey.Usage.UNWRAP;
+            usages[2] = SymmetricKey.Usage.ENCRYPT;
+            usages[3] = SymmetricKey.Usage.DECRYPT;
+
+            kg.setKeyUsages(usages);
+            kg.temporaryKeys(true);
+            // Handle 128 and 256 initialization sizes
+            kg.initialize(256);
+            SymmetricKey tempKey = kg.generate();
+
+            // Now wrap and unwrap with AES CBC PAD
+
+            KeyWrapper keyWrap = token.getKeyWrapper(KeyWrapAlgorithm.AES_CBC_PAD);
+            int ivLen = KeyWrapAlgorithm.AES_CBC_PAD.getBlockSize();
+            byte[] iv = new byte[ivLen];
+
+            IVParameterSpec ivsp = new IVParameterSpec(iv);
+            keyWrap.initWrap(tempKey, ivsp);
+            byte [] wrapped = keyWrap.wrap(aesKey);
+
+            Cipher decryptor = token.getCipherContext(EncryptionAlgorithm.AES_256_CBC_PAD);
+            decryptor.initDecrypt(tempKey,ivsp);
+            result = decryptor.doFinal(wrapped);
+
+         } catch (Exception e) {
+             return result;
+         }
+
+         return result;
+     }
+     
+      private boolean checkAllowDebugKeyRollover() {
+        boolean allow = true;
+
+        String method = "SecureChannelProtocol.checkAllowDebugKeyRollover: ";
+        IConfigStore cs = CMS.getConfigStore();
+
+        String allowDebugKeyRollover = "tks.debugKeyRollover";
+
+        CMS.debug(method + " trying config: " + allowDebugKeyRollover);
```

Comment:
I suggest that you comment this out.  We don't need to advertise this parameter.

Reply:
Will do.